### PR TITLE
Make the analyzer plugin actor a status client actor

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This makes it possible to receive status information from analyzer plugin actors.

Here's how it looks when it works:

```json
❯ vast -qq status | jq .importer
{
  "analyzers": [
    {
      "example": {
        "answer": 42
      }
    }
  ]
}
```

And here's a fabricated test case with a timeout:

```json
❯ vast -qq status | jq .importer
{
  "analyzers": [
    {
      "example": {
        "error": "!! request_timeout"
      }
    }
  ]
}
```

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

**Review by mavam — COMMENTED** [review]
Overall, this seems fine. I left some smaller comments in the review.

Have you tested this with two analyzer plugins as well?

File: libvast/vast/system/importer.hpp [hunk 1]
```diff
@@ -75,14 +76,17 @@ struct importer_state {
...
+  caf::typed_response_promise<caf::settings> status(status_verbosity v) const;
 
   /// The active id block.
   id_block current;
 
   /// State directory.
   path dir;
 
+  /// All available ANALYZER PLUGIN actors and their names.
+  std::vector<std::pair<std::string, analyzer_plugin_actor>> analyzers;
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/vast/system/importer.hpp:88]: Don't think so right now, they're local to the importer. We can always change that later.

**Comment by dominiklohmann** [comment]
> Have you tested this with two analyzer plugins as well?

Yes, locally with the inventory plugin. We can't really have that in CI right now, but it works as expected.

File: libvast/src/system/importer.cpp [hunk 2]
```diff
@@ -196,7 +204,36 @@ importer_state::status(status_verbosity v) const {
...
+          auto& analyzer_status = caf::put_dictionary(
+            analyzers_status.emplace_back().as_dictionary(), name);
+          caf::put(analyzer_status, "error", render(err));
+          if (--req_state->pending_replies == 0)
+            rp.deliver(std::move(req_state->result));
+        });
+  }
+  if (req_state->pending_replies == 0)
+    rp.deliver(std::move(req_state->result));
+  return rp;
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/system/importer.cpp:236]: We have the same pattern elsewhere, let's not discuss this now. This is a more general issue with status requests, and we should first try to get rid of this cascading logic entirely.

File: examples/plugins/example/example.cpp [hunk 3]
```diff
@@ -53,36 +54,46 @@ example(example_actor::stateful_pointer<example_actor_state> self) {
...
-          }
-        },
-        // Teardown hook for CAF stram.
-        [=](uint64_t&, const caf::error& err) {
-          if (err && err != caf::exit_reason::user_shutdown) {
-            VAST_ERROR("{} finished stream with error: {}", self, render(err));
-            return;
-          }
-        });
+      return caf::attach_stream_sink(
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply examples/plugins/example/example.cpp:60]: Nope. :/

**Review by mavam — APPROVED** [review]

**Pull request merged**